pkg/apis/rollouts/v1alpha1/analysis_types.go

```diff
@@ -316,6 +331,38 @@ type AnalysisRunStatus struct {
 	MetricResults []MetricResult `json:"metricResults,omitempty" protobuf:"bytes,3,rep,name=metricResults"`
 	// StartedAt indicates when the analysisRun first started
 	StartedAt *metav1.Time `json:"startedAt,omitempty" protobuf:"bytes,4,opt,name=startedAt"`
+	// RunSummary contains the final status of the metric results collected from the run
+	RunSummary RunSummary `json:"runSummary,omitempty" protobuf:"bytes,5,opt,name=runSummary"`
+}
+
+// RunSummary contains the final results from the metric executions
+type RunSummary struct {
+	// This is equal to the sum of Successful, Failed, Inconclusive
+	Count int32 `json:"count,omitempty" protobuf:"varint,1,opt,name=count"`
+	// Successful is the number of times the metric was measured Successful
+	Successful int32 `json:"successful,omitempty" protobuf:"varint,2,opt,name=successful"`
+	// Failed is the number of times the metric was measured Failed
+	Failed int32 `json:"failed,omitempty" protobuf:"varint,3,opt,name=failed"`
+	// Inconclusive is the number of times the metric was measured Inconclusive
+	Inconclusive int32 `json:"inconclusive,omitempty" protobuf:"varint,4,opt,name=inconclusive"`
+	// Error is the number of times an error was encountered during measurement
+	Error int32 `json:"error,omitempty" protobuf:"varint,5,opt,name=error"`
+	// DryRun contains the final status of the metric results collected from the dry-run mode
+	DryRun DryRunSummary `json:"dryRun,omitempty" protobuf:"bytes,6,opt,name=dryRun"`
+}
+
+// DryRunSummary contains the final results from the metric executions in the dry-run mode
+type DryRunSummary struct {
+	// This is equal to the sum of Successful, Failed, Inconclusive
+	Count int32 `json:"count,omitempty" protobuf:"varint,1,opt,name=count"`
+	// Successful is the number of times the metric was measured Successful
+	Successful int32 `json:"successful,omitempty" protobuf:"varint,2,opt,name=successful"`
+	// Failed is the number of times the metric was measured Failed
+	Failed int32 `json:"failed,omitempty" protobuf:"varint,3,opt,name=failed"`
+	// Inconclusive is the number of times the metric was measured Inconclusive
+	Inconclusive int32 `json:"inconclusive,omitempty" protobuf:"varint,4,opt,name=inconclusive"`
+	// Error is the number of times an error was encountered during measurement
+	Error int32 `json:"error,omitempty" protobuf:"varint,5,opt,name=error"`
```

Comment:
Instead of embedding dryRunSummary into runSummary i think this will be cleaner:

1) Have a single RunSummary type which re-used for both `status.runSummary` and `status.dryRunSummary`

```golang
	// RunSummary contains the final status of the metric results collected from the run
	RunSummary RunSummary `json:"runSummary,omitempty" protobuf:"bytes,5,opt,name=runSummary"`

	// DryRunSummary contains the final status of the metric results collected from the run
	DryRunSummary *RunSummary `json:"dryRunSummary,omitempty" protobuf:"bytes,6,opt,name=runSummary"`
```

Also, `dryRunSummary` should be a pointer because it will be omitted in the case that dryRun is not used.


```yaml
status:
  summary:
    count: 1
    ...
  dryRunSummary:
    count: 1
    ...
```

Reply:
Changed, Thanks!